File: eTravelAgencija.Model/model/PaymentSummary.cs
using System;

namespace eTravelAgencija.Model.model
{
    public class PaymentSummary
    {
        // ======================
        // I RATA
        // ======================
        public bool IsFirstRateVisible { get; set; }
        public bool IsFirstRateDisabled { get; set; }
        public bool? IsFirstRatePending { get; set; }
        // null  -> ne postoji zapis
        // false -> postoji zapis, isConfirmed = false
        // true  -> postoji zapis, isConfirmed = true

        // ======================
        // II RATA
        // ======================
        public bool IsSecondRateVisible { get; set; }
        public bool IsSecondRateDisabled { get; set; }
        public bool? IsSecondRatePending { get; set; }

        // ======================
        // III RATA
        // ======================
        public bool IsThirdRateVisible { get; set; }
        public bool IsThirdRateDisabled { get; set; }
        public bool? IsThirdRatePending { get; set; }

        // ======================
        // PREOSTALI IZNOS
        // ======================
        public bool IsRemainingVisible { get; set; }
        public bool IsRemainingDisabled { get; set; }
        public bool? IsRemainingPending { get; set; }

        // ======================
        // PUNI IZNOS
        // ======================
        public bool IsFullAmountVisible { get; set; }
        public bool IsFullAmountDisabled { get; set; }
        public bool? IsFullAmountPending { get; set; }

        public DateTime? PaymentDeadline {get; set;}
    }
}

File: UI/etravel_mobile/lib/models/payment_summary.dart
class PaymentSummary {
  // ======================
  // I RATA
  // ======================
  final bool isFirstRateVisible;
  final bool isFirstRateDisabled;
  final bool? isFirstRatePending;

  // ======================
  // II RATA
  // ======================
  final bool isSecondRateVisible;
  final bool isSecondRateDisabled;
  final bool? isSecondRatePending;

  // ======================
  // III RATA
  // ======================
  final bool isThirdRateVisible;
  final bool isThirdRateDisabled;
  final bool? isThirdRatePending;

  // ======================
  // PREOSTALI IZNOS
  // ======================
  final bool isRemainingVisible;
  final bool isRemainingDisabled;
  final bool? isRemainingPending;

  // ======================
  // PUNI IZNOS
  // ======================
  final bool isFullAmountVisible;
  final bool isFullAmountDisabled;
  final bool? isFullAmountPending;

  final String? paymentDeadline;

  PaymentSummary({
    required this.isFirstRateVisible,
    required this.isFirstRateDisabled,
    required this.isFirstRatePending,

    required this.isSecondRateVisible,
    required this.isSecondRateDisabled,
    required this.isSecondRatePending,

    required this.isThirdRateVisible,
    required this.isThirdRateDisabled,
    required this.isThirdRatePending,

    required this.isRemainingVisible,
    required this.isRemainingDisabled,
    required this.isRemainingPending,

    required this.isFullAmountVisible,
    required this.isFullAmountDisabled,
    required this.isFullAmountPending,

    required this.paymentDeadline
  });

  factory PaymentSummary.fromJson(Map<String, dynamic> json) {
    return PaymentSummary(
      isFirstRateVisible: json['isFirstRateVisible'] as bool,
      isFirstRateDisabled: json['isFirstRateDisabled'] as bool,
      isFirstRatePending: json['isFirstRatePending'] as bool?,

      isSecondRateVisible: json['isSecondRateVisible'] as bool,
      isSecondRateDisabled: json['isSecondRateDisabled'] as bool,
      isSecondRatePending: json['isSecondRatePending'] as bool?,

      isThirdRateVisible: json['isThirdRateVisible'] as bool,
      isThirdRateDisabled: json['isThirdRateDisabled'] as bool,
      isThirdRatePending: json['isThirdRatePending'] as bool?,

      isRemainingVisible: json['isRemainingVisible'] as bool,
      isRemainingDisabled: json['isRemainingDisabled'] as bool,
      isRemainingPending: json['isRemainingPending'] as bool?,

      isFullAmountVisible: json['isFullAmountVisible'] as bool,
      isFullAmountDisabled: json['isFullAmountDisabled'] as bool,
      isFullAmountPending: json['isFullAmountPending'] as bool?,

      paymentDeadline: json['paymentDeadline'] as String? 
    );
  }
}
